src/command-palette/types.d.ts

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+/**
+ * Command configuration interface.
+ */
+export interface CommandConfig {
+	name: string;
+	label: string;
+	icon?: React.ReactNode;
+	callback: ( options: { close: () => void } ) => void;
+	context?: string;
+}
+
+/**
+ * Command loader configuration interface.
+ */
+export interface CommandLoaderConfig {
+	name: string;
+	hook: ( params: { search: string } ) => {
+		commands: CommandConfig[];
+		isLoading: boolean;
+	};
+}
+
+/**
+ * ActivityPub Command Palette configuration passed from PHP.
+ */
+export interface ActivityPubCommandPaletteConfig {
+	followingEnabled: boolean;
+	actorMode: 'actor' | 'blog' | 'actor_blog';
```

Comment:
Just a note: We should find a way, to share consts between PHP and JS. The direct use of the values is a bit fragile.